tests/test_config.py

```diff
@@ -1,10 +1,44 @@
 import unittest
 from mock import Mock
-from microproxy.config import Config
+from microproxy.config import Config, ConfigParserBuilder
 
 
 class ConfigTest(unittest.TestCase):
-    def test_proxy_cmd_options(self):
+    def test_cmd_proxy_arguments(self):
+        parser = ConfigParserBuilder.setup_cmd_parser()
+        arguments = [
+            "proxy",
+            "--host", "127.0.0.1",
+            "--port", "8080",
+            "--mode", "socks",
+            "--http-port", "5000,5001",
+            "--https-port", "5002,5003",
+            "--viewer-channel", "tcp://*:5581"
+        ]
+        config = vars(parser.parse_args(arguments))
+
+        assert config["command_type"] == "proxy"
+        assert config["host"] == "127.0.0.1"
+        assert config["port"] == 8080
+        assert config["mode"] == "socks"
+        assert config["http_port"] == "5000,5001"
+        assert config["https_port"] == "5002,5003"
+        assert config["viewer_channel"] == "tcp://*:5581"
+
+    def test_cmd_viewer_arguments(self):
+        parser = ConfigParserBuilder.setup_cmd_parser()
+        arguments = [
+            "viewer",
+            "--viewer-mode", "log",
+            "--viewer-channel", "tcp://127.0.0.1:5581"
+        ]
+        config = vars(parser.parse_args(arguments))
+
+        assert config["command_type"] == "viewer"
+        assert config["viewer_mode"] == "log"
+        assert config["viewer_channel"] == "tcp://127.0.0.1:5581"
```

Comment:
If we took command line argument as our setting, I think we need to check the value and throw error when it's illegal.